### PR TITLE
Use ES6 class declarations + lint

### PR DESCRIPTION
**Pull request by vidartf**
Since the code is run through TS compilation, I figured we might as well use a more modern variant of JS. Also did some automatic linting.

Mostly did it to make it easier for myself to read the code :)

### CODE REVIEW

**Comment by maartenbreddels** [comment]
Great work! Will need to go over it carefully, did you only 'polish' the code, no bugfixes etc?

File: js/src/webrtc.js [hunk 1]
```diff
@@ -4,36 +4,42 @@ require('webrtc-adapter');
...
+        };
+    }
+
+    get stream() {
+        return this.captureStream();
+    }
+
+    captureStream() {
+        throw new Error('Not implemented');
+    }
```
Comment (vidartf) [inline js/src/webrtc.js]: I adder this just to indicate that this is an abstract base class, where this method should be overloaded.

**Comment by vidartf** [comment]
Except for the one change noted above, everything else should be refactoring. That said, some of the refactoring is significant, e.g. changing to arrow functions (and thereby changing `that`for `this`); using const/let instead of var, which implicitly changes scoping of some variables; using `super`, which might have been significant if you were purposely skipping steps in the inheritance chain. I'm not sure if the tests give high enough coverage for me to be 100% confident that I didn't break anything.

**Comment by maartenbreddels** [comment]
Thanks a lot for the work! I pushed two fixes, the rest seems to work. Do you want me to merge this?

**Comment by vidartf** [comment]
If it works, then by all means :)

**Comment by maartenbreddels** [comment]
Again, thanks for the massive amount of work!

**Comment by vidartf** [comment]
Mostly smart use of search and replace ;)

**Pull request merged**